clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -397,8 +397,12 @@ public synchronized void closeStorePartition(VeniceStoreVersionConfig storeConfi
   public synchronized void removeStorageEngine(String kafkaTopic) {
     AbstractStorageEngine<?> storageEngine = getStorageEngineRepository().removeLocalStorageEngine(kafkaTopic);
     if (storageEngine == null) {
-      LOGGER.warn("Storage engine {} does not exist, ignoring remove request.", kafkaTopic);
-      return;
+      LOGGER.warn("Storage engine {} does not exist, trying to reopen storage for .", kafkaTopic);
+      storageEngine = openStorageEngine(kafkaTopic);
```

Comment:
This also calls `addLocalStorageEngine` which needs to be removed similar to line 398.